<?xml version="1.0"?>
<launch>
    <!-- Name of car -->
    <arg name="vehicle_name" default="SVEA3"/>

    <!-- coordinate parameters for the path -->
    <arg name="coords_param" default= '[[-1.55, 3.77, -2.3],
                                        [-5.82, -1.19, -2.1],
                                        [-5.6, -4.28, -0.63],
                                        [-3.47, -5.93, 0.12],
                                        [-1.25, -5.37, 0.8978652],
                                        [0.39, -3.14, 0.8978652],
                                        [0.48, -0.4, 1.25],
                                        [1.35, 1.18, 0.8978652],
                                        [5.23, 3.22, 0.8978652],
                                        [8.3, 6.25, 0.8978652],
                                        [10.24, 8.6, 0.94],
                                        [11.1, 10.7, 1.67],
                                        [9.77, 12.67, 2.3],
                                        [6.76, 13.08, -2.3],
                                        [2.31, 8.2, -2.3]]'/>
    <arg name="curvature" default="0.82" />
    

    <arg name="long_look_ahead" default="0.6" />
    <arg name="short_look_ahead" default="0.3" />

    <!-- Simulation parameters -->
    <arg name="use_rviz" default="false"/>
    <arg name="use_matplotlib" default="false"/>

    <!-- Map parameters -->
    <arg name="map_file" default="$(find svea)/src/scripts/maps/floor2.simplemap" />

    <!-- Speed limit -->
    <arg name="speed_limit" default="2.5" />

    <!-- Gear -->
    <arg name="gear" default="1" />


    <!-- launch rviz -->
    <group if="$(eval use_rviz == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea)/src/scripts/rviz/SVEA_indoor_team3.rviz" />
    </group>

    <!-- get map -->
    <node pkg="mrpt_map" type="map_server_node" name="map_server_node" output="screen">
        <param name="debug" value="false"/>
        <param name="frame_id" value="map"/>
        <param name="frequency" value="1"/>
        <param name="ini_file" value="$(find svea_sensors)/params/mrpt/pf_localization.ini"/>
        <param name="map_file" value="$(arg map_file)"/>
    </node>

    <!-- Initialize the sensors -->
    <include file="$(find svea_sensors)/launch/localize.launch" />

    <!-- Initialize low level control connection -->
    <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
        <param name="port" value="/dev/arduinoPWM"/>
        <param name ="baud" value="250000" />
        <remap from="/lli/ctrl_request" to="/SVEA3/lli/ctrl_request" />
        <remap from="/lli/ctrl_actuated" to="/SVEA3/lli/ctrl_actuated" />
    </node>

    <!-- Emergency breaking node -->
    <node pkg="svea" type="emergency_brake.py" name="emergency_brake" output="screen">
        <param name="vehicle_name" value="$(arg vehicle_name)"/>
    </node>

    <!-- Transform to map -->
        <node pkg="svea" type="transform_to_map.py" name="transform_to_map" output="screen">
    </node>

    <!-- launch pure pursuit on floor 2 -->
    <node pkg="svea" type="floor2_pure_pursuit.py" name="floor2_pure_pursuit" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="coords_param" value="$(arg coords_param)"/>
        <param name="curvature" value="$(arg curvature)"/>
        <param name="speed_limit" value="$(arg speed_limit)"/>
        <param name="gear" value="$(arg gear)"/>
        <param name="long_look_ahead" value="$(arg long_look_ahead)"/>
        <param name="short_look_ahead" value="$(arg short_look_ahead)"/>
    </node>
</launch>
